utils/python/CIME/SystemTests/eri.py

```diff
@@ -2,16 +2,221 @@
 CIME ERI test  This class inherits from SystemTestsCommon
 """
 from CIME.XML.standard_module_setup import *
-from system_tests_common import SystemTestsCommon
+from CIME.SystemTests.system_tests_common import SystemTestsCommon
+from CIME.utils import run_cmd
+
+import shutil, glob, os
+
+logger = logging.getLogger(__name__)
 
 class ERI(SystemTestsCommon):
 
-    def __init__(self, caseroot=None, case=None):
+    def __init__(self, case):
         """
         initialize an object interface to the ERI system test
         """
-        SystemTestsCommon.__init__(self, caseroot=caseroot, case=case)
+        SystemTestsCommon.__init__(self, case)
         self._testname = "ERI"
 
     def run(self):
-        SystemTestsCommon.run(self)
+        caseroot = self._case.get_value("CASEROOT")
+        clone1_path = "%s.ref1" % caseroot
+        clone2_path = "%s.ref2" % caseroot
+        #self._case.set_value("CHECK_TIMING", False)
+
+        #
+        # clone the main case to create ref1 and ref2 cases
+        #
+        clones = []
+        for clone_path in [clone1_path, clone2_path]:
+            if os.path.exists(clone_path):
+                shutil.rmtree(clone_path)
+
+            clones.append(self._case.create_clone(clone_path, keepexe=True))
+
+        clone1, clone2 = clones
+        orig_case = self._case
+        orig_casevar = orig_case.get_value("CASE")
+
+        #
+        # determine run lengths needed below
+        #
+        stop_n = self._case.get_value("STOP_N")
+        stop_option = self._case.get_value("STOP_OPTION")
+        run_startdate = self._case.get_value("RUN_STARTDATE")
+
+        stop_n1 = stop_n / 6
+        rest_n1 = stop_n1
+        start_1 = run_startdate
+
+        stop_n2 = stop_n - stop_n1
+        rest_n2 = stop_n2 / 2 + 1
+        hist_n  = stop_n2
+
+        start_1_year, start_1_month, start_1_day = [int(item) for item in start_1.split("-")]
+        start_2_year = start_1_year + 2
+        start_2 = "%04d-%02d-%02d" % (start_2_year, start_1_month, start_1_day)
+
+        stop_n3 = stop_n2 - rest_n2
+        rest_n3 = stop_n3 / 2 + 1
+
+        stop_n4 = stop_n3 - rest_n3
+
+        expect(stop_n4 >= 1 and stop_n1 >= 1, "Run length too short")
+
+        #
+        # (1) Test run:
+        # do an initial ref1 case run
+        # cloned the case and running there
+        # (NOTE: short term archiving is on)
+        #
+
+        os.chdir(clone1_path)
+        self._set_active_case(clone1)
+
+        logger.info("ref1 startup: doing a %s %s startup run from %s and 00000 seconds" % (stop_n1, stop_option, start_1))
+        logger.info("  writing restarts at %s %s" % (rest_n1, stop_option))
+        logger.info("  short term archiving is on ")
+
+        clone1.set_value("CONTINUE_RUN", False)
```

Comment:
If self active case is clone1 why do you use clone1 here instead of self?  Is it the same thing?